theano/gpuarray/subtensor.py

```diff
@@ -20,6 +20,22 @@
                         infer_context_name)
 
 
+iadd_reg = {}
+
+
+def get_iadd(a, b):
+    key = (a.type.dtype, b.type.dtype, a.type.context)
+    if key not in iadd_reg:
+        if a.dtype == 'float16' or b.dtype == 'float16':
+            raise NotImplementedError('float16 is not supported by pygpu '
+                                      'elemwise')
```

Comment:
So this mean that for GpuIncSubtensor, this PR introduce a regression, the perform was supporting float16, but now it isn't.

This also mean that GpuAdvancedIncSubtensor1 wasn't supporting float16, while the issue told it was.

Do you agree with both statement? How long it would be to add this to remove the regression?